### PR TITLE
Improved view in table

### PR DESCRIPTION
**Pull request by mateusz-git**

### CODE REVIEW

**Comment by RafalProoo** [comment]
There is a problem with setting warning text (NPE).

File: src/main/java/application/Controller.java [hunk 1]
```diff
@@ -204,17 +204,33 @@ public void onClickCheckProductWholesalePrice() {
 
     public ObservableList<DataInTable> getDataForAllStateList(State state, List<State> stateList, ProductInfo productInfo, double price) {
         ObservableList<DataInTable> dataInTableObservableList = FXCollections.observableArrayList();
+        int numberOfStateWhereYouLostMoney = 0;
+
```
Comment (GGuinea) [inline src/main/java/application/Controller.java:208]: remove new line

File: src/main/java/application/Controller.java [hunk 2]
```diff
@@ -232,18 +248,20 @@ protected void updateItem(String item, boolean empty) {
                 TableRow<DataInTable> currentRow = getTableRow();
 
                 if (!isEmpty()) {
-                    if (item.equals(state.getState())) {
-                        currentRow.setStyle("-fx-background-color:lightgreen");
+                    if (item.charAt(0) == '-') {
+                        currentRow.setStyle("-fx-background-color:lightcoral");
+                    } else if (item.length() == 1 && item.charAt(0) == '0') {
```
Comment (GGuinea) [inline src/main/java/application/Controller.java:253]: "1" could be extracted to local variable, it is not clear why should be equal to 1?

File: src/test/java/application/ControllerTests.java [hunk 3]
```diff
@@ -10,21 +10,23 @@
...
+    private State alabama = new State("Alabama", 4, taxesInAlabama, 3);
 
     @Test
     public void calculateWithoutTax_ReturnedSpecifiedValue() {
         double userPrice = 10;
         double groceriesTax = 1;
 
         double priceWithoutTax = controller.calculateWithoutTax(userPrice, apple, alaska);
-        double expectedPriceWithouTax = userPrice / ((100 + groceriesTax) / 100);
+        double expectedPriceWithoutTax = userPrice / ((100 + groceriesTax) / 100);
```
Comment (GGuinea) [inline src/test/java/application/ControllerTests.java:27]: "100" could be extracted to variable, sounds like magic number

File: src/main/java/application/Controller.java [hunk 4]
```diff
@@ -232,18 +248,20 @@ protected void updateItem(String item, boolean empty) {
...
+                        currentRow.setStyle("-fx-background-color:yellow");
                     } else {
-                        currentRow.setStyle("-fx-background-color:white");
-                        currentRow.setStyle("-fx-text-fill: black");
+                        currentRow.setStyle("-fx-background-color:lightgreen");
+                        //currentRow.setStyle("-fx-text-fill: black");
```
Comment (GGuinea) [inline src/main/java/application/Controller.java:257]: Comments should be removed from production code

File: src/main/java/application/Controller.java [hunk 5]
```diff
@@ -204,17 +204,33 @@ public void onClickCheckProductWholesalePrice() {
...
-                dataInTableObservableList.add(new DataInTable(s.getState(), formatPrice(priceWithoutTax), formatPrice(margin), String.valueOf(s.getLogisticCosts())));
+                marginString = formatPrice(margin).equals("-0") ? "0" : formatPrice(margin);
+                if (Double.parseDouble(marginString) < 0) {
+                    numberOfStateWhereYouLostMoney++;
+                }
+                dataInTableObservableList.add(new DataInTable(s.getState(), formatPrice(priceWithoutTax), marginString, String.valueOf(s.getLogisticCosts())));
             }
         }
+        if (numberOfStateWhereYouLostMoney > 0) {
+            if (numberOfStateWhereYouLostMoney == 1)
```
Comment (GGuinea) [inline src/main/java/application/Controller.java:229]: Code formatter should add here brackets

File: src/main/java/application/Controller.java [hunk 6]
```diff
@@ -204,17 +204,33 @@ public void onClickCheckProductWholesalePrice() {
 
     public ObservableList<DataInTable> getDataForAllStateList(State state, List<State> stateList, ProductInfo productInfo, double price) {
         ObservableList<DataInTable> dataInTableObservableList = FXCollections.observableArrayList();
+        int numberOfStateWhereYouLostMoney = 0;
+
         double priceWithoutTax = calculateWithoutTax(price, productInfo, state);
         double margin = calculateMargin(priceWithoutTax, productInfo, state.getLogisticCosts());
-        dataInTableObservableList.add(new DataInTable(state.getState(), formatPrice(priceWithoutTax), formatPrice(margin), String.valueOf(state.getLogisticCosts())));
+        String marginString = formatPrice(margin).equals("-0") ? "0" : formatPrice(margin);
+        if (Double.parseDouble(marginString) < 0) {
```
Comment (GGuinea) [inline src/main/java/application/Controller.java:212]: Ok, so we can accept oneliner's without brackets, but we should stay consistent with whole file at least

**Pull request merged**